src/nifc/codegen.nim

```diff
@@ -350,11 +350,24 @@ proc genVarDecl(c: var GeneratedCode; t: Tree; n: NodePos; vk: VarKind; toExtern
     if vis == StaticC:
       c.code.insert(Token(StaticKeyword), beforeDecl)
     if t[d.value].kind != Empty:
-      c.add AsgnOpr
-      if vk != IsLocal: inc c.inSimpleInit
-      genx c, t, d.value
-      if vk != IsLocal: dec c.inSimpleInit
-    c.add Semicolon
+      if vk == IsGlobal and
+        t[d.value].kind notin {IntLit, UIntLit, FloatLit, CharLit, StrLit, FalseC, TrueC}:
```

Comment:
But "bracket expressions" for arrays and objects are fine too and it would be wrong to move them!